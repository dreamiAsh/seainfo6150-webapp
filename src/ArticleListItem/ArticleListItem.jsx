import React from "react";
import styles from "./ArticleListItem.module.css";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import ArticleImage from "../ArticleImage/ArticleImage"

const ArticleListItem = (props) => {
  return (
    <article className={styles.box} onClickCapture={()=> {}}>
      <ArticleImage url={props.article.image._url} title={props.article.title}/>
      <h1 className={styles.header}>{props.article.title}</h1>
      <ArticleTextToggleButton date={props.article.displayDate} shortText={props.article.shortText}/>
    </article>
  );
};

export default ArticleListItem;
