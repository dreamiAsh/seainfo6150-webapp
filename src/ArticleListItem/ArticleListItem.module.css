.box{
    background: white;
    margin-right: 10px;
    text-decoration: underline;
    text-decoration-color: red;
    border-radius: 10px;
    padding: 2px 5px 13px 20px;
    margin-bottom: 20px;
    background-image: radial-gradient(circle at -100%, red,  #eee 75%, #eee 75%);
    border: solid #b04d4d 1px;
    box-shadow: 5px 10px 10px #676f6f ;
}
.header{
    color: #cd3b27;
    height: 30px;
}