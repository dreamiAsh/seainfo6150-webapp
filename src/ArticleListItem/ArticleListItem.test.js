import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {
  it("renders correctly", () => {
    const article = {"title": "a", "displayDate": "b", "shortText": "c"};
    const { container } = render(<ArticleListItem article={article}/>);
    expect(container).toMatchSnapshot();
  });
});