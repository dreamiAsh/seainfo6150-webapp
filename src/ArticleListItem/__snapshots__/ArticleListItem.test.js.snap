// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <article
    class="box"
  >
    <h1
      class="header"
    >
      a
    </h1>
    <div>
      <button
        class="button"
      >
        Show more
      </button>
    </div>
  </article>
</div>
`;
